token/cli/src/main.rs

```diff
@@ -1377,13 +1377,22 @@ fn command_close(
         }
     }
 
-    let instructions = vec![close_account(
+    let mut instructions = vec![close_account(
         &config.program_id,
         &account,
         &recipient,
         &close_authority,
         &config.multisigner_pubkeys,
     )?];
+
+    let recipient_account = config
+        .rpc_client
+        .get_token_account(&recipient)?;
+
+    if !recipient_account.is_none() && recipient_account.unwrap().is_native {
+        instructions.push(sync_native(&config.program_id, &recipient)?);
+    }
+
```

Comment:
These checks need to be moved up in the `if !config.sign_only` block, since in offline mode, we can't fetch the recipient account.